### ISSUE TITLE
Project Has Conflicts with ``Black``

### ISSUE THREAD

**Issue by acederberg** [shamrin/codeblocks#2]
Hi!

I wanted to play with this but there are a few issues. I am using ``poetry`` to manage my dependencies in my project and wanted to include this like

```sh
poetry add codeblocks
```

However it looks like the ``click`` version used here conflicts with that used for ``black``:

```stdout
Using version ^0.2.3 for codeblocks

Updating dependencies
Resolving dependencies... (0.0s)

Because no versions of codeblocks match >0.2.3,<0.3.0
 and codeblocks (0.2.3) depends on click (>=7.0,<8.0), codeblocks (>=0.2.3,<0.3.0) requires click (>=7.0,<8.0).
And because black (24.10.0) depends on click (>=8.0.0)
 and no versions of black match >24.10.0,<25.0.0, codeblocks (>=0.2.3,<0.3.0) is incompatible with black (>=24.10.0,<25.0.0).
So, because quarto-demo depends on both black (^24.10.0) and codeblocks (^0.2.3), version solving failed.
```

I will see what I can do about this and possibly make a PR.

**Comment by acederberg** [comment]
It also looks like [typed_ast](https://github.com/python/typed_ast) has hit its end of life. For me, this means in this project cloned I get a nasty error when I run

```sh
poetry install
```

**Comment by acederberg** [comment]
See #3 for fixes.

``click`` had to be updated a major version since ``typed_ast`` is dead.

@shamrin have you ever used ``pandoc``? My usecase is that I have ``quarto`` documents that I want to format. ``quarto`` is basically extended ``pandoc`` markdown - there is no complete tool for fixing quarto documents (except in neovim, which requires quite a bit of setup).

I noticed that there is the task to choose a markdown parser on the todo list, but this might not work across the various flavors of markdown, so it might be optimal to stick to using regex.

**Comment by shamrin** [comment]
Thank you for the bug report! I haven’t used pandoc much. But it seems to support markdown style triple backticks too? I won’t be surprised if `codeblocks` would already work with them too.

https://pandoc.org/chunkedhtml-demo/8.5-verbatim-code-blocks.html

**Comment by shamrin** [comment]
I have also updated TODO.md file to make it clear that switching to Markdown parser is just a possible idea, not fully decided thing.

**Comment by shamrin** [comment]
@acederberg I've published codeblocks version 0.3.0 that includes these fixes. I've also updated README.md to show more ways to install it. Thanks again for help!
